runtime/lua/vim/treesitter.lua

```diff
@@ -438,11 +426,14 @@ end
 ---@param lang (string|nil) Language of the parser (default: from buffer filetype)
 function M.start(bufnr, lang)
   bufnr = bufnr or api.nvim_get_current_buf()
-  local parser = M._get_parser(bufnr, lang)
-  if not parser then
-    vim.notify('No parser for the given buffer.', vim.log.levels.WARN)
-    return
-  end
+  local parser = assert(
+    M.get_parser(bufnr, lang, { error = false }),
+    string.format(
```

Comment:
I think that is a LuaJIT-only signature. (And constructing the error message inside an `assert` is exactly what we shouldn't do -- we should just use the message that `get_parser` constructs in the fail branch so we can just `assert` it.)

Reply:
Oh, wait, `bufnr` and especially `lang` are only determined in `get_parser`. Then we _have_ to use the `nil, errmsg` pattern and just `assert(get_parser(...))` here!